### ISSUE TITLE
Fix Lint Warnings in the Repository

### ISSUE THREAD

**Issue by shubhanshu02** [COPS-IITBHU/hackalog-frontend#131]
### Description

This issue deals with fixing the lint errors in the Components throughout the repository.

### Output 

The following errors can be seen in the output of `npm run lint` on the repository;

```
/hackalog-frontend/components/Profile/EditProfileModal.tsx
  85:25  warning  'e' is defined but never used  no-unused-vars

/hackalog-frontend/context/auth.tsx
   11:21  warning  Caution: `React` also has a named export `createContext`. Check if you meant to write `import {createContext} from 'react'` instead  import/no-named-as-default-member
   34:45  warning  Caution: `React` also has a named export `useState`. Check if you meant to write `import {useState} from 'react'` instead            import/no-named-as-default-member
   35:31  warning  Caution: `React` also has a named export `useState`. Check if you meant to write `import {useState} from 'react'` instead            import/no-named-as-default-member
   36:35  warning  Caution: `React` also has a named export `useState`. Check if you meant to write `import {useState} from 'react'` instead            import/no-named-as-default-member
   37:35  warning  Caution: `React` also has a named export `useState`. Check if you meant to write `import {useState} from 'react'` instead            import/no-named-as-default-member
   48:21  warning  'error' is defined but never used                                                                                                    no-unused-vars
   64:21  warning  'err' is defined but never used                                                                                                      no-unused-vars
   80:30  warning  'err' is defined but never used                                                                                                      no-unused-vars
  102:37  warning  'error' is defined but never used                                                                                                    no-unused-vars
  106:29  warning  'error' is defined but never used                                                                                                    no-unused-vars
  113:5   warning  Caution: `React` also has a named export `useEffect`. Check if you meant to write `import {useEffect} from 'react'` instead          import/no-named-as-default-member
  115:8   warning  React Hook React.useEffect has a missing dependency: 'checkAuth'. Either include it or remove the dependency array                   react-hooks/exhaustive-deps
  136:30  warning  Caution: `React` also has a named export `useContext`. Check if you meant to write `import {useContext} from 'react'` instead        import/no-named-as-default-member

/hackalog-frontend/context/firebase.ts
   2:8   warning  '/hackalog-frontend/node_modules/firebase/empty-import.d.ts' imported multiple times                       import/no-duplicates
   3:8   warning  '/hackalog-frontend/node_modules/firebase/empty-import.d.ts' imported multiple times                       import/no-duplicates
  20:10  warning  Caution: `firebase` also has a named export `apps`. Check if you meant to write `import {apps} from 'firebase/app'` instead                    import/no-named-as-default-member
  21:9   warning  Caution: `firebase` also has a named export `initializeApp`. Check if you meant to write `import {initializeApp} from 'firebase/app'` instead  import/no-named-as-default-member
  22:9   warning  Caution: `firebase` also has a named export `analytics`. Check if you meant to write `import {analytics} from 'firebase/app'` instead          import/no-named-as-default-member
  30:9   warning  Caution: `firebase` also has a named export `auth`. Check if you meant to write `import {auth} from 'firebase/app'` instead                    import/no-named-as-default-member
  36:21  warning  'err' is defined but never used                                                                                                                no-unused-vars

/hackalog-frontend/pages/contributors.tsx
  123:8   warning  React Hook useEffect has a missing dependency: 'fullname'. Either include it or remove the dependency array. You can also do a functional update 'setFullName(f => ...)' if you only need 'fullname' in the 'setFullName' call  react-hooks/exhaustive-deps
  165:58  warning  'index' is defined but never used                                                                                                                                                                                               no-unused-vars

/hackalog-frontend/pages/hackathon/[slug]/index.tsx
   83:25  warning  'err' is defined but never used                                                                                                                                                                                                                                                                                                                                          no-unused-vars
   88:8   warning  React Hook useEffect has a missing dependency: 'hackathon'. Either include it or remove the dependency array                                                                                                                                                                                                                                                             react-hooks/exhaustive-deps
  560:25  warning  'err' is defined but never used                                                                                                                                                                                                                                                                                                                                          no-unused-vars

/hackalog-frontend/pages/hackathon/[slug]/register.tsx
  11:32  warning  'TeamSerializer' is defined but never used  no-unused-vars

/hackalog-frontend/pages/hackathon/[slug]/teams/[team_id]/index.js
  4:8  warning  Using exported name 'Router' as identifier for default export  import/no-named-as-default

/hackalog-frontend/pages/hackathon/[slug]/teams/[team_id]/submit.tsx
    9:5   warning  'FormControlProps' is defined but never used                                                                                                                                                                                                                                                                                                                             no-unused-vars
   12:10  warning  'ChangeEvent' is defined but never used                                                                                                                                                                                                                                                                                                                                  no-unused-vars
   23:10  warning  'AxiosError' is defined but never used                                                                                                                                                                                                                                                                                                                                   no-unused-vars
   64:21  warning  Unexpected console statement                                                                                                                                                                                                                                                                                                                                             no-console
   69:8   warning  React Hook useEffect has a missing dependency: 'token'. Either include it or remove the dependency array                                                                                                                                                                                                                                                                 react-hooks/exhaustive-deps

/hackalog-frontend/pages/hackathons.tsx
  30:21  warning  'err' is defined but never used  no-unused-vars
  43:21  warning  'err' is defined but never used  no-unused-vars
  53:21  warning  'err' is defined but never used  no-unused-vars

/hackalog-frontend/pages/index.tsx
    2:8   warning  'Header' is defined but never used                                                                                           no-unused-vars
   25:41  warning  Caution: `React` also has a named export `useState`. Check if you meant to write `import {useState} from 'react'` instead    import/no-named-as-default-member
   28:35  warning  Caution: `React` also has a named export `useState`. Check if you meant to write `import {useState} from 'react'` instead    import/no-named-as-default-member
   29:5   warning  Caution: `React` also has a named export `useEffect`. Check if you meant to write `import {useEffect} from 'react'` instead  import/no-named-as-default-member
   37:17  warning  Unexpected console statement                                                                                                 no-console

/hackalog-frontend/pages/profile/[username].tsx
  85:21  warning  Unexpected console statement  no-console
```

### Contribution

New contributors are welcome to put up pull requests. To avoid multiple people working on the same file, please comment below and get yourself assigned your choice from the below.

**Note**: For completing the firebase related part of this issue, you may refer to [HacktoBoard Repository's `Context` Folder](https://github.com/shubhanshu02/hacktoboard/tree/main/src/context)

- [x] General Files and Profile Page: (#133)

  - /hackalog-frontend/pages/index.tsx
  - /hackalog-frontend/pages/hackathons.tsx
  - /hackalog-frontend/pages/contributors.tsx
  - /hackalog-frontend/pages/profile/[username].tsx
  - /hackalog-frontend/components/Profile/EditProfileModal.tsx

- [x] Authentication Context (#143)

  - /hackalog-frontend/context/auth.tsx
  - /hackalog-frontend/context/firebase.ts

- [x] Hackathon Specific Pages (#140)

  - /hackalog-frontend/pages/hackathon/[slug]/index.tsx
  - /hackalog-frontend/pages/hackathon/[slug]/register.tsx
  - /hackalog-frontend/pages/hackathon/[slug]/teams/[team_id]/submit.tsx
  - /hackalog-frontend/pages/hackathon/[slug]/teams/[team_id]/index.js





PS: In order to accommodate more new contributors, we are limiting the count to one choice per contributor.

**Comment by anil-muraleedharan** [comment]
I would like to take this issue :)

**Comment by anil-muraleedharan** [comment]
General Files and Profile Page part. 

Some small clarifications
- I hope the console.error is also supposed to be removed
- The useEffect hook dependency array empty is not a lint warning (I mean it might be needed in the code) [/hackalog-frontend/pages/contributors.tsx]

**Comment by shubhanshu02** [comment]
> * I hope the console.error is also supposed to be removed

Yep, right.

> * The useEffect hook dependency array empty is not a lint warning (I mean it might be needed in the code) [/hackalog-frontend/pages/contributors.tsx]

Since `fullname` is being updated here from the previous state, it's better to use a [functional update](https://reactjs.org/docs/hooks-reference.html#functional-updates) here. If you have some other suggestion, feel free to discuss here.

**Comment by shubhanshu02** [comment]
@AjayPrabuR, here are some notes that may help,

For the first one,
- The Effect hook here sets up an observer (`firebase.auth().onAuthStateChanged`) on the Authentication state of the firebase user. So whenever the user logs in, the user details are fetched from the backend and sets the states in the context.
- Since this part is required only once when the page loads, the dependency array was kept empty.
- ESLint shows the warning that `checkAuth` function needs to be in the dependency array. 
- For now, we can put up the code in the function directly in the Effect hook since the `checkAuth` function isn't being used anywhere else. But if you have a better solution, please let us know.

For the second one, 
- I guess it's better to remove those import comments since they do not serve any purpose. 
- The auth and analytics need to be imported since removing them with the current code structure gives an error: `TypeError: firebase.auth is not a function`.
- You might want to have a look at how authentication code is structured [here](https://github.com/shubhanshu02/hacktoboard/tree/main/src/context).

**Comment by Pravithaop** [comment]
I would like to take Hackathon Specific Pages part in this issue.

I’m not able to fix this specific lint error below , it would be great if you could help me with this.

  /hackalog-frontend/pages/hackathon/[slug]/teams/[team_id]/index.js
  4:8  warning  Using exported name 'Router' as identifier for default export  import/no-named-as-default

**Comment by Pravithaop** [comment]
@shubhanshu02 Sure, thanks for the clarifications. Will fix the issues and raise a PR

**Comment by AjayPrabuR** [comment]
@shubhanshu02 Hi, based on the suggestion for changes I have raised a new PR

**Comment by ankur12-1610** [comment]
> That's already merged @ankur12-1610

Oh!
Sorry, I didn't notice it.
